index.js

```diff
@@ -1,13 +1,72 @@
 /* jshint node: true */
 'use strict';
-var SSHAdapter = require('./lib/ssh-adapter');
+var DeployPluginBase = require('ember-cli-deploy-plugin');
+var path       = require('path');
 
 module.exports = {
-  name: 'ember-deploy-ssh-index',
-  type: 'ember-deploy-addon',
-  adapters: {
-    index: {
-      'ssh': SSHAdapter
-    }
+  name: 'ember-cli-deploy-ssh-index',
+
+  createDeployPlugin: function(options) {
+    var SSHAdapter = require('./lib/ssh-adapter');
+
+    var DeployPlugin = DeployPluginBase.extend({
+      name: options.name,
+
+      defaultConfig: {
+        revisionKey: function(context) {
+          return context.commandOptions.revision || (context.revisionData && context.revisionData.revisionKey);
+        },
+        keyPrefix: function(context) {
+          return context.project.name();
+        },
+        filePattern: 'index.html',
+        distDir: function(context) {
+          return context.distDir;
+        },
+        didDeployMessage: function(context) {
+          var revisionKey = context.revisionData && context.revisionData.revisionKey;
+          var activatedRevisionKey = context.revisionData && context.revisionData.activatedRevisionKey;
+          if (revisionKey && !activatedRevisionKey) {
+            return "Deployed but did not activate revision " + revisionKey + ". "
+                 + "To activate, run: "
+                 + "ember deploy:activate " + context.deployTarget + " --revision=" + revisionKey + "\n";
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>Bad line breaking before '+'.